### ISSUE TITLE
Why we need new layer 2 protocol!!

### ISSUE THREAD

**Issue by OmidHekayati** [GeniusesGroup/memar#1]
We believe that layer 2 Ethernet doesn't fit for the next generation of networks with billions of billion devices!
In almost every switch device(layer 2) of a network, a switch must store a state as mac to port! and also in layer 3, every router must store additional state as IP to mac to have the ability to route packets!
In [introduced protocol](https://github.com/SabzCity/RFCs/blob/master/Chapar.md) we want to change this behavior to have two stateful layers in a network! I know you want to say now that IP is not stateful but please think about this, We talk about state in devices, not protocol header here!!
Eliminating state from devices in layer 2 have some goals:
- A huge decrease in a network set up the price!
- A huge increase in the speed of layer 2 networks due to having very simple stateless devices!! Switch devices don't need any RAM type eg. CAM, ... to store state
- A huge decrease in power usage due to having a much much simpler device than existing layer 2 devices!!
- A huge increase in layer 2 capacity! the introduced network can setup for 2^2048 devices without the need for the upper layer to route data!

In best and most expensive exiting switch devices like [this](https://www.cisco.com/c/en/us/products/collateral/switches/nexus-9000-series-switches/datasheet-c78-742284.html) with more than 30k$, you can set up a network with max 512,000 nodes! And almost you must get help from the upper layer to set up a reliable network with this number of nodes!

The Introduces switch protocol can implement in devices with 256 port by very low price and very high throughput that can't compare to existing devices like [this](https://www.edge-core.com/productsInfo.php?cls=1&cls2=158&cls3=159&id=299)!

In [IoE - IoT era](https://en.wikipedia.org/wiki/Internet_of_things) if we want all devices can communicate directly to each other without help from other networks like bus, we need to update layer 2 protocol, What do you think?

**Comment by OmidHekayati** [comment]
This issue was opened in 2020 as a holding place for *why* Ethernet’s layer-2 state model does not fit a network of that scale, and what a replacement would buy. That question is no longer waiting on a later answer. The current specification is [`docs/protocols/chapar.md`](https://github.com/GeniusesGroup/memar/blob/dev/docs/protocols/chapar.md).

What this issue asked for is now stated there as a protocol, not as a sketch:

- Intermediate switches hold no MAC-to-port (or any per-address) table. Forwarding knowledge lives in the frame — the hop-port sequence — and at the two peers, never in the switches in between. See [State model compared with Ethernet](https://github.com/GeniusesGroup/memar/blob/dev/docs/protocols/chapar.md#state-model-compared-with-ethernet). The comparison is about *where devices hold state*, which is the distinction this issue already insisted on.
- The cost, power, simplicity, and scale goals (no CAM, cheaper switching silicon, a much larger usable layer 2) are the protocol’s [Goals](https://github.com/GeniusesGroup/memar/blob/dev/docs/protocols/chapar.md#goals-and-non-goals).
- [Discovery](https://github.com/GeniusesGroup/memar/blob/dev/docs/protocols/chapar.md#discovery) is how a device obtains a path before its first Unicast, without Ethernet’s ARP plus learned tables.

A few 2020 numbers were tightened rather than copied:

- `2^2048` devices with no layer 3 is not a claim the spec still makes. Capacity is topology-shaped ([topology examples](https://github.com/GeniusesGroup/memar/blob/dev/docs/protocols/chapar.md#topology-capacity-examples)); header-budget comparisons are illustrative (`255^16` vs Ethernet’s `2^48`). Both protocols still hand off to layer 3 past a practical layer-2 diameter. Chapar’s difference is that it reaches that point without CAM in the switches.
- Device-to-device communication without a bus overlay is in scope; composing a path between two devices still uses ChaparKhane as coordinator, not a fully coordinator-free mesh.
- The ~512,000 MAC-table ceiling is kept as an order-of-magnitude figure pending a sourced product citation. That is documentation hygiene, not an open “why Chapar” question.

Remaining Chapar follow-ups (Discovery-response authenticity, a ChaparKhane path-directory document, an empirical header comparison) live in the protocol handoff, not here.
